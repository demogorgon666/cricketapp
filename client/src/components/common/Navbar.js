import React, { Component } from "react";
import "../css/Navbar.css";
import { Link, Redirect } from "react-router-dom";
import { getSearch } from "../../actions/SerachAction";
import { connect } from "react-redux";
import decode from "jwt-decode";

let decoded_token;

export class Navbar extends Component {
	state = {
		active: true,
		searchInput: "",
		isChecked: window.innerWidth >= 526 ? true : false,
		width: window.innerWidth,
		redirect: false,
		pageLink: this.props.pageLink ? this.props.pageLink : "",
		showGender: this.props.showGender
	};

	toggleChange = pageLink => {
		this.setState({ pageLink });
		window.innerWidth <= 526
			? this.setState({ isChecked: !this.state.isChecked })
			: console.log("do nothing");
	};

	handleSearchInputChange = e => {
		this.setState({ [e.target.name]: e.target.value });
		let search_term = {
			search: this.state.searchInput
		};
		this.props.getSearch(search_term, this.props.history);
	};

	componentDidMount() {
		window.addEventListener("resize", this.updateDimensions);
		if (localStorage.getItem("token")) {
			decoded_token = decode(localStorage.getItem("token"));
		}
	}

	updateDimensions = () => {
		this.setState({
			width: window.innerWidth,
			isChecked: window.innerWidth >= 526 ? true : false
		});
	};

	renderRedirect = () => {
		if (this.state.redirect) {
			return <Redirect to="/login" />;
		}
	};

	logout = () => {
		localStorage.removeItem("token");
		this.setState({ redirect: true });
	};

	render() {
		return (
			<div className="nav-parent">
				{this.renderRedirect()}
				<nav className="menu">
					<ul>
						{/* brand name which is at left of navbar */}
						<li className="nav-brand-li">
							<Link
								to="/"
								id="nav-brand"
								onClick={() => {
									this.setState({ pageLink: "" });
								}}
							>
								CricketAlpha
							</Link>
						</li>

						{/* toggle gender buttons for men and women */}
						<div
							className="genders"
							style={{
								display: this.state.showGender
									? "default"
									: "none"
							}}
						>
							<span
								id="men"
								className={
									this.props.gender === "male"
										? "active-gender"
										: ""
								}
								onClick={() => this.props.changeGender("male")}
							>
								Men
							</span>
							<span
								id="women"
								className={
									this.props.gender === "female"
										? "active-gender"
										: ""
								}
								onClick={() =>
									this.props.changeGender("female")
								}
							>
								Women
							</span>
						</div>

						{/* hidden checkbox to show and hide the menu in small screen device */}
						<input
							type="checkbox"
							id="menu-btn"
							className="menu-btn"
							checked={this.state.isChecked}
							onClick={() =>
								this.toggleChange(this.state.pageLink)
							}
						/>

						{/* label for checkbox which makes the hamburger menu */}
						<label
							className="menu-icon"
							for="menu-btn"
							style={{
								display:
									this.state.width < 526 ? "block" : "none"
							}}
						>
							<span className="bars"></span>
						</label>

						{/* matches link */}
						<li
							className={
								this.state.pageLink == "matches"
									? "item active-link"
									: "item"
							}
							style={{
								display: this.state.isChecked ? "block" : "none"
							}}
							onClick={() => this.toggleChange("matches")}
						>
							<Link className="link" to="/matches" id="matches">
								Matches
							</Link>
						</li>

						{/* teams link */}
						<li
							className={
								this.state.pageLink == "teams"
									? "item active-link"
									: "item"
							}
							style={{
								display: this.state.isChecked ? "block" : "none"
							}}
							onClick={() => this.toggleChange("teams")}
						>
							<Link className="link" to="/teams" id="teams">
								Teams
							</Link>
						</li>

						{/* players link */}
						<li
							className={
								this.state.pageLink == "players"
									? "item active-link"
									: "item"
							}
							style={{
								display: this.state.isChecked ? "block" : "none"
							}}
							onClick={() => this.toggleChange("players")}
						>
							<Link className="link" to="/players" id="players">
								Players
							</Link>
						</li>

						{/* search box list item */}
						<li className="search-box-li">
							{/* search box to search for players and teams */}
							<input
								id="searchInput"
								name="searchInput"
								className="search-box"
								value={this.state.searchInput}
								onChange={this.handleSearchInputChange}
								placeholder="Search for Team or Player"
							/>

							{/* shows search result after entering text in search boc */}

							<div
								className="search-result"
								style={{
									display: this.state.searchInput
										? "block"
										: "none"
								}}
							>
								{/* checks whether search has entries or not 
								if not then prints no result found */}
								{this.props.search.length !== 0 ? (
									this.props.search.player.length !== 0 ||
									this.props.search.team.length !== 0 ? (
										<>
											{/* checks for player array length in seaarch
										if present map all the player available */}
											{this.props.search.player.length !==
											0
												? this.props.search.player.map(
														mapped_search => (
															<div>
																<Link
																	className="search-list-item"
																	to={{
																		pathname:
																			"/playerInfo/" +
																			mapped_search.player_id
																		// state:{
																	}}
																>
																	<div
																		style={{
																			backgroundImage: `url(data:image/jpeg;base64,${mapped_search.player_image}`
																		}}
																	></div>
																	<div
																		className="search-suggestion"
																		id={
																			"player-search-" +
																			mapped_search.player_id
																		}
																	>
																		{
																			mapped_search.player_name
																		}
																	</div>
																</Link>
															</div>
														)
												  )
												: null}

											{/* checks for team array length in seaarch
										if present map all the team available */}
											{this.props.search.team.length !== 0
												? this.props.search.team.map(
														mapped_search => (
															<div>
																<Link
																	className="search-list-item"
																	to={{
																		pathname:
																			"/teamdetails/" +
																			mapped_search.team_id
																		// state:{
																	}}
																>
																	<div
																		style={{
																			backgroundImage: `url(data:image/jpeg;base64,${mapped_search.team_image}`
																		}}
																	></div>
																	<div
																		className="search-suggestion"
																		id={
																			"team-search-" +
																			mapped_search.team_id
																		}
																	>
																		{
																			mapped_search.team_name
																		}
																	</div>
																</Link>
															</div>
														)
												  )
												: null}
										</>
									) : (
										<div className="search-no-result">
											<div style={{ color: "#272727" }}>
												No result found
											</div>
										</div>
									)
								) : null}
							</div>
						</li>

						{/* icon for login */}
						<li
							className="login-icon"
							style={{
								display: this.state.isChecked ? "block" : "none"
							}}
						>
							<Link className="link" to="/login">
								<i className="fas fa-user"></i>
							</Link>
							<div className="user-options-links">
								{!localStorage.getItem("token") ? (
									<>
										<div>
											<Link
												className="user-options-item"
												to="/login"
											>
												<div id="login">Login</div>
											</Link>
										</div>
										<div>
											<Link
												className="user-options-item"
												to="/register"
											>
												<div id="register">
													Register
												</div>
											</Link>
										</div>
									</>
								) : decoded_token ? (
									decoded_token.isAdmin ? (
										<>
											<div>
												<Link
													className="user-options-item"
													to="/adminteam"
												>
													<div id="manage-teams">
														Manage Teams
													</div>
												</Link>
											</div>
											<div>
												<Link
													className="user-options-item"
													to="/adminplayer"
												>
													<div id="manage-players">
														Manage Players
													</div>
												</Link>
											</div>
											<div>
												<Link
													className="user-options-item"
													onClick={this.logout}
												>
													<div id="logout-admin">
														Logout
													</div>
												</Link>
											</div>
										</>
									) : (
										<div>
											<Link
												className="user-options-item"
												onClick={this.logout}
											>
												<div id="logout">Logout</div>
											</Link>
										</div>
									)
								) : null}
							</div>
						</li>
					</ul>
				</nav>
			</div>
		);
	}
}

const mapStateToProps = state => ({
	search: state.SearchReducer.search
});

export default connect(
	mapStateToProps,
	{ getSearch }
)(Navbar);
