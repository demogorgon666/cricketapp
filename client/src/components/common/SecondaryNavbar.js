import React, { Component } from "react";
import "../css/SecondaryNavbar.css";
import dhoni from "../images/dhoni.jpg";
import { Link } from "react-router-dom";

export default class SecondaryNavbar extends Component {
	render() {
		return (
			<div className="secNavParent">
				<div className="secNavHead">
					<div
						style={{
							backgroundImage: `url(${dhoni})`
						}}
					></div>
					<div>
						<h2>MS Dhoni</h2>
						<h4>India</h4>
					</div>
				</div>
				<div className="secNavLinks">
					<ul>
						<Link className="secLink" to="/info">
							<li>Info</li>
						</Link>
						<Link className="secLink" to="/battingStats">
							<li>Batting Stats</li>
						</Link>
						<Link className="secLink" to="/bowlingStats">
							<li>Bowling Stats</li>
						</Link>
						<Link className="secLink" to="/performance">
							<li>Performance</li>
						</Link>
					</ul>
				</div>
			</div>
		);
	}
}
