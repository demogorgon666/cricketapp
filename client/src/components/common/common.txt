includes common files for components