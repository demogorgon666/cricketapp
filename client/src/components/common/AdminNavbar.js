import React, { Component } from "react";
import "../css/Navbar.css";
import { Link } from "react-router-dom";
import { getSearch } from "../../actions/SerachAction";
import { connect } from "react-redux";

export class AdminNavbar extends Component {
	state = {
		active: true,
		searchInput: "",
		isChecked: window.innerWidth >= 526 ? true : false,
		width: window.innerWidth
	};

	toggleChange = () => {
		this.setState({ isChecked: !this.state.isChecked });
	};

	componentDidMount() {
		window.addEventListener("resize", this.updateDimensions);
	}

	updateDimensions = () => {
		this.setState({
			width: window.innerWidth,
			isChecked: window.innerWidth >= 526 ? true : false
		});
	};

	render() {
		return (
			<div className="nav-parent">
				<nav className="menu admin">
					<ul className="admin-ul">
						{/* brand name which is at left of navbar */}
						<li className="nav-brand-li admin-nav-brand">
							<Link to="/" id="nav-brand">
								CricketAlpha
							</Link>
						</li>

						{/* hidden checkbox to show and hide the menu in small screen device */}
						<input
							type="checkbox"
							id="menu-btn"
							className="menu-btn"
							checked={this.state.isChecked}
							onClick={this.toggleChange}
							// hidden="hidden"
						/>

						{/* label for checkbox which makes the hamburger menu */}
						<label
							className="menu-icon"
							for="menu-btn"
							style={{
								display:
									this.state.width < 526 ? "block" : "none"
							}}
						>
							<span className="bars"></span>
						</label>

						{/* manage matches link */}
						<li
							className="item"
							style={{
								display: this.state.isChecked ? "block" : "none"
							}}
							onClick={this.toggleChange}
						>
							<Link
								className="link"
								to="/adminplayer"
								id="adminplayer"
							>
								Manage Players
							</Link>
						</li>

						{/* manage teams link */}
						<li
							className="item"
							style={{
								display: this.state.isChecked ? "block" : "none"
							}}
							onClick={this.toggleChange}
						>
							<Link
								className="link"
								to="/adminteam"
								id="adminteam"
							>
								Manage Teams
							</Link>
						</li>
					</ul>
				</nav>
			</div>
		);
	}
}

const mapStateToProps = state => ({
	search: state.SearchReducer.search
});

export default connect(
	mapStateToProps,
	{ getSearch }
)(AdminNavbar);
