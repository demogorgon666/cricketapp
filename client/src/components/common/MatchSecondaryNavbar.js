import React, { Component } from "react";
import "../css/MatchSecondaryNavbar.css";
import "../css/SecondaryNavbar.css";
import south_africa from "../images/SouthAfrica.jpeg";
import india from "../images/india.jpeg";
export default class MatchSecondaryNavbar extends Component {
  state = {
    summary: true,
    scorecard: false,
    stats: false
  }
  render() {
    return (
      <div>
        <div className="secNavParent">
          <div className="matchSecNavHead">
            <div className="match-score-details">
              <span id="match-result" style={{ fontSize: "14px" }}>
                Result:{" "}
                <b id="match-winner-name-runs">
                  {this.props.match.team_winner} {this.props.match.won_by}
                </b>
              </span>
              <div className="match-team-score">
                <div>
                  <div
                    style={{
                      backgroundImage: `url(data:image/jpeg;base64,${this.props.match.team_one_img}`
                    }}
                  ></div>
                  <div className="team-name">{this.props.match.teamOne}</div>
                  <div className="team-score">
                    {" "}
                    {this.props.match.teamOneScore}/
                    {this.props.match.teamone_wicket} (
                    {this.props.match.team_one_total_over}.{this.props.match.team_one_total_ball} overs)
                  </div>
                </div>
                <div>
                  <div
                    style={{
                      backgroundImage: `url(data:image/jpeg;base64,${this.props.match.team_two_img}`
                    }}
                  ></div>
                  <div className="team-name">{this.props.match.teamTwo}</div>
                  <div className="team-score">
                    {" "}
                    {this.props.match.teamTwoScore}/
                    {this.props.match.teamtwo_wicket} (
                    {this.props.match.team_two_total_over}.{this.props.match.team_two_total_ball} overs)
                  </div>
                </div>
              </div>
            </div>
            <div className="match-pom">
              <span style={{ fontSize: "14px" }}>Player of the match</span>
              <div className="pom-name-photo">
                <div className="pom-img">
                  <i className="fas fa-user"></i>
                </div>
                <div className="pom-name">
                  <div>{this.props.match.player_of_the_match}</div>
                  <div>{this.props.match.team_winner}</div>
                </div>
              </div>
            </div>
          </div>
          <div className="secNavLink">
            <ul>
              <span
                id="summary-click"
                className="secLink"
                onClick={() => {
                  this.setState({ summary: true, scorecard: false, stats: false })
                  this.props.changeDetailsType("summary");
                }}
              >
                <li id="match-summary" className={this.state.summary ? "options-selected-li" : "options-li"}>Summary</li>
              </span>
              <span
                className="secLink"
                onClick={() => {
                  this.setState({ summary: false, scorecard: true, stats: false })
                  this.props.changeDetailsType("scorecard");
                }}
              >
                <li id="match-scorecard" className={this.state.scorecard ? "options-selected-li" : "options-li"}>ScoreBoard</li>
              </span>
              <span
                className="secLink"
                onClick={() => {
                  this.setState({ summary: false, scorecard: false, stats: true })
                  this.props.changeDetailsType("stats");
                }}
              >
                <li id="match-stats" className={this.state.stats ? "options-selected-li" : "options-li"}>Statistics</li>
              </span>
            </ul>
          </div>
        </div>
      </div>
    );
  }
}
