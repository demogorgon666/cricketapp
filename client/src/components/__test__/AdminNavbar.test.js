import React from "react";
import { shallow, mount } from "enzyme";
import { AdminNavbar } from "../common/AdminNavbar";

const toggleChange = jest.fn();
const updateDimensions = jest.fn();
const wrapper = shallow(
	<AdminNavbar
		toggleChange={toggleChange}
		updateDimensions={updateDimensions}
	/>
);

describe("test for the text, input, css properties and icons on the navbar and snapshot testing", () => {
	it("renders the component for snapshot testing", () => {
		expect(wrapper).toMatchSnapshot();
	});

	it("should have text brand CricketAlpha, matches, teams, players", () => {
		// checks for the links text
		expect(wrapper.find("#adminteam").text()).toBe("Manage Teams");
		expect(wrapper.find("#adminplayer").text()).toBe("Manage Players");

		// checks for the nav brand text
		expect(wrapper.find("#nav-brand").text()).toBe("CricketAlpha");
	});
});

describe("test for toggle on navbar and toggleChange function", () => {
	it("checks for the toggleChange function to be called when menu-btn is clicked", () => {
		// expects toggleChange to be called on clicking men link
		wrapper.find("#menu-btn").simulate("click");
		expect(toggleChange).toHaveBeenCalled();
	});
});

describe("test for updateDimensions function to be called", () => {
	it("checks for the updateDimensions function to be called on window size change", () => {
		const spy = jest.spyOn(wrapper.instance(), "updateDimensions");

		// Trigger the window resize event.
		global.addEventListener("resize", spy);
		global.dispatchEvent(new Event("resize"));
		// expects updateDimensions to be called on clicking men link
		// wrapper.find("#menu-btn").simulate("click");
		expect(spy).toBeCalled();
	});
});
