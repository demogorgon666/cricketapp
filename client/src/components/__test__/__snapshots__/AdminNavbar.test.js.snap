// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for the text, input, css properties and icons on the navbar and snapshot testing renders the component for snapshot testing 1`] = `
<div
  className="nav-parent"
>
  <nav
    className="menu admin"
  >
    <ul
      className="admin-ul"
    >
      <li
        className="nav-brand-li admin-nav-brand"
      >
        <Link
          id="nav-brand"
          to="/"
        >
          CricketAlpha
        </Link>
      </li>
      <input
        checked={true}
        className="menu-btn"
        id="menu-btn"
        onClick={[Function]}
        type="checkbox"
      />
      <label
        className="menu-icon"
        for="menu-btn"
        id="menu-btn-label"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <span
          className="bars"
        />
      </label>
      <li
        className="admin-item"
        onClick={[Function]}
        style={
          Object {
            "display": "block",
          }
        }
      >
        <Link
          className="link"
          id="adminplayer"
          to="/adminplayer"
        >
          Manage Players
        </Link>
      </li>
      <li
        className="admin-item"
        onClick={[Function]}
        style={
          Object {
            "display": "block",
          }
        }
      >
        <Link
          className="link"
          id="adminteam"
          to="/adminteam"
        >
          Manage Teams
        </Link>
      </li>
      <li
        className="admin-item"
        onClick={[Function]}
        style={
          Object {
            "display": "block",
          }
        }
      >
        <Link
          className="link"
          id="logout-admin-page"
        >
          Logout
        </Link>
      </li>
    </ul>
  </nav>
</div>
`;
